python-checks/src/main/java/org/sonar/python/checks/cdk/UnrestrictedAdministrationCheck.java

```diff
@@ -0,0 +1,230 @@
+
+  private static final String IP_PROTOCOL = "ip_protocol";
+  private static final String CIDR_IP = "cidr_ip";
+  private static final String CIDR_IPV6 = "cidr_ipv6";
+  private static final String IPPROTOCOL = "ipProtocol";
+  private static final String CIDRIP = "cidrIp";
+  private static final String CIDRIPV6 = "cidrIpv6";
+
+
+  private static final Set<String> BAD_PROTOCOL = Set.of("tcp", "6");
+  private static final String INVALID_PROTOCOL = "-1";
+  private static final String EMPTY_IPV4 = "0.0.0.0/0";
+  private static final String EMPTY_IPV6 = "::/0";
+  private static final long[] ADMIN_PORTS = new long[]{22, 3389};
+
+  private SubscriptionContext ctx;
+
+  @Override
+  protected void registerFqnConsumer() {
+    checkFqn("aws_cdk.aws_ec2.CfnSecurityGroup", this::checkCfnSecurityGroup);
+    checkFqn("aws_cdk.aws_ec2.CfnSecurityGroupIngress", this::checkCfnSecurityGroupIngress);
+  }
+
+  // Checks methods
+  private void checkCfnSecurityGroup(SubscriptionContext ctx, CallExpression callExpression) {
+    this.ctx = ctx;
+
+    getArgument(ctx, callExpression, "security_group_ingress")
+      .flatMap(flow -> flow.getExpression(isListLiteral()).map(ListLiteral.class::cast))
+      .ifPresent(list -> list.elements().expressions()
+        // Process each expression in the Python list
+        .forEach(expression -> {
+          // Process each location of those expressions
+          CdkUtils.ExpressionFlow flow = CdkUtils.ExpressionFlow.build(ctx, expression);
+          for (Expression expr : flow.locations()) {
+            raiseIssueIfIngressPropertyCallWithSensitiveArgument(expr);
+            raiseIssueIfDictionaryWithSensitiveArgument(expr);
+          }
+        })
+      );
+  }
+
+  private void checkCfnSecurityGroupIngress(SubscriptionContext ctx, CallExpression callExpression) {
+    this.ctx = ctx;
+
+    if (isBadProtocolWithEmptyIpAddressAndAdminPort(callExpression) || isInvalidProtocolWithEmptyIpAddress(callExpression)) {
+      getArgument(ctx, callExpression, CIDR_IP).ifPresent(flow -> flow.addIssue(MESSAGE));
+      getArgument(ctx, callExpression, CIDR_IPV6).ifPresent(flow -> flow.addIssue(MESSAGE));
+    }
+  }
+
+  // Methods to handle call expressions
+  private void raiseIssueIfIngressPropertyCallWithSensitiveArgument(Expression expression) {
+    getCall(expression, "aws_cdk.aws_ec2.CfnSecurityGroup.IngressProperty")
+      .filter(callExpression -> isBadProtocolWithEmptyIpAddressAndAdminPort(callExpression) || isInvalidProtocolWithEmptyIpAddress(callExpression))
+      .ifPresent(callExpression -> {
+        getArgument(ctx, callExpression, CIDR_IP).ifPresent(flow -> flow.addIssue(MESSAGE));
+        getArgument(ctx, callExpression, CIDR_IPV6).ifPresent(flow -> flow.addIssue(MESSAGE));
+      });
+  }
+
+  private boolean isBadProtocolWithEmptyIpAddressAndAdminPort(CallExpression callExpression) {
+    return is(callExpression, IP_PROTOCOL, isString(BAD_PROTOCOL))
```

Comment:
The name of the method ìs` should be more explicit about what actually is checked. Maybe something like ìsCallWithArgument`. Or you could think about a small class like `Call` which wrapped the `CallExpression` and has a method `boolean hasArgument(String arg, Predicate<Expression> predicate)`. This class could also contain the ctx, which is needed for the resolution of the argument.

Reply:
Good idea for the Call class, I will add it !